ext/travisci/prep-os-essentials-for

```diff
@@ -11,6 +11,21 @@ misuse() {
     exit 2
 }
 
+# this handles the issue where apt-get update fails but has an exit code of zero
+# because there's a mirror sync going on. This failure is retryable, but you have
+# to look for it in the output from the command. This is a common failure on AWS
+# and GCP because they inject their own mirrors into /etc/apt/sources.list at VM
+# start time, and there's a delay in getting all their mirrors in sync.
+apt_get_update() {
```

Comment:
Let's use
```
    local tmpdir
    tmpdir=$(mktemp)
```
perhaps just right before the `for` in the function (need two lines because local hides "set -e" failures)